cmd/segment-reaper/detect.go

```diff
@@ -212,14 +243,89 @@ func cmdDetect(cmd *cobra.Command, args []string) (err error) {
 	log.Info("number of inline segments", zap.Int("segments", observer.inlineSegments))
 	log.Info("number of last inline segments", zap.Int("segments", observer.lastInlineSegments))
 	log.Info("number of remote segments", zap.Int("segments", observer.remoteSegments))
+	log.Info("number of all segments", zap.Int("segments", observer.remoteSegments+observer.inlineSegments))
 	return nil
 }
 
 func analyzeProject(ctx context.Context, db metainfo.PointerDB, objectsMap ObjectsMap, csvWriter *csv.Writer) error {
-	// TODO this part will be implemented in next PR
+	for cluster, objects := range objectsMap {
+		for path, object := range objects {
+			if object.skip {
+				continue
+			}
+
+			segments := make([]byte, 0)
+			for i := byte(0); i < maxNumOfSegments; i++ {
+				found := object.segments&(1<<uint64(i)) != 0
+				if found {
+					segments = append(segments, i)
+				}
+			}
+
+			if object.hasLastSegment {
+				brokenObject := false
+				if object.expectedNumberOfSegments == 0 {
+					for i := 0; i < len(segments); i++ {
+						if int(segments[i]) != i {
+							brokenObject = true
+							break
+						}
+					}
+				} else if len(segments) != int(object.expectedNumberOfSegments)-1 {
+					// expectedNumberOfSegments-1 because 'segments' doesn't contain last segment
+					brokenObject = true
+				}
+
+				if !brokenObject {
+					segments = []byte{}
+				} else {
+					err := printSegment(ctx, db, cluster, "l", path, csvWriter)
+					if err != nil {
+						return err
+					}
+				}
+			}
+
+			for _, segmentIndex := range segments {
+				err := printSegment(ctx, db, cluster, "s"+strconv.Itoa(int(segmentIndex)), path, csvWriter)
```

Comment:
Why don't we pass the segment index as an `int` and let the `printSegment` function do the `"s"+strconv.Itoa(int(segmentIndex))` formatting?

Reply:
because then we need separate logic to handle the last segment. I'm ok with both versions and can change this if you think it will be cleaner.